### PR TITLE
Refactor PeriodicWave ctor.

### LINKED ISSUE #1244: Define`createPeriodicWave` in terms of the `PeriodicWave` ctor and not the opposite
We now do this for `AudioNode`s, it makes much more sense.

### PR DESCRIPTION
**Pull request by padenot**
This fixes #1244.

### CODE REVIEW

**Comment by padenot** [comment]
Preview: https://rawgit.com/padenot/web-audio-api/2457160107cd383d48ef907139a9e28e1087572a/index.html

File: index.html [hunk 1]
```diff
@@ -10390,15 +10373,84 @@ <h2 id="WaveShaperNode">
...
+              internal slots of type <code>Float32Array</code>, of length both
+              equal to the maximum length of the <var>real</var> and
+              <var>imag</var> of the attributes of the
+              <a>PeriodicWaveOptions</a> passed in. <a href=
+              "https://heycam.github.io/webidl/#dfn-get-buffer-source-copy">Make
+              a copy</a> of those arrays into their respective internal slots.
+            </li>
+            <li>
+              Let [[normalize]] be an internal slot of the <a>PeriodicWave</a>
+              the is initialize to the inverse of the
```
Comment (rtoy) [inline index.html]: s/initialize/initialized/

File: index.html [hunk 2]
```diff
@@ -10390,15 +10373,84 @@ <h2 id="WaveShaperNode">
...
+                BaseAudioContext context
+              </dt>
+              <dd>
+                <p>
+                  The <a>BaseAudioContext</a> for which to create this
+                  <a>PeriodicWave</a>.
+                </p>
+                <p>
+                  Unlike <a>AudioBuffer</a>, <a>PeriodicWave</a>s can't be
+                  shared accross <a>AudioContext</a> or
```
Comment (rtoy) [inline index.html]: s/accross/across an/

File: index.html [hunk 3]
```diff
@@ -10439,23 +10491,34 @@ <h2 id="WaveShaperNode">
...
             <dd>
-              The array of <code>cosine</code> terms equivalent to the
-              <a><code>real</code></a> parameter to <a href=
-              "#widl-BaseAudioContext-createPeriodicWave-PeriodicWave-Float32Array-real-Float32Array-imag-PeriodicWaveConstraints-constraints">
-              <code>createPeriodicWave</code></a>. This defaults to a sequence
-              of all zeroes of the same length as <a><code>imag</code></a> if
-              <a><code>imag</code></a> is given.
+              <p>
+                The <dfn id="dfn-real">real</dfn> parameter represents an array
+                of <code>cosine</code> terms (traditionally the A terms). In
```
Comment (rtoy) [inline index.html]: I think it's best to leave off the "(traditionally the A terms)".  Or perhaps better just say this represents the a[k] terms in the waveform generation section.

Likewise for the B terms below when discussing the imag element.

File: index.html [hunk 4]
```diff
@@ -10439,23 +10491,34 @@ <h2 id="WaveShaperNode">
...
-              The array of <code>sine</code> terms terms equivalent to the
-              <a><code>imag</code></a> parameter to <a href=
-              "#widl-BaseAudioContext-createPeriodicWave-PeriodicWave-Float32Array-real-Float32Array-imag-PeriodicWaveConstraints-constraints">
-              <code>createPeriodicWave</code></a>. This defaults to a sequence
-              of all zeroes of the same length as <a><code>real</code></a> if
-              <a><code>real</code></a> is given.
+              <p>
+                The <dfn id="dfn-imag">imag</dfn> parameter represents an array
+                of <code>sine</code> terms (traditionally the B terms). The
+                first element (index 0) should be set to zero (and will be
```
Comment (rtoy) [inline index.html]: I think you should make "and will be ignored" part written consistently for both real and imag elements. So make the both parenthetical as here or both complete sentences as done above for the real element.

**Comment by padenot** [comment]
Update with you comments addressed.

File: index.html [hunk 5]
```diff
@@ -10511,12 +10505,11 @@ <h2 id="WaveShaperNode">
...
               <p>
                 The <dfn id="dfn-imag">imag</dfn> parameter represents an array
-                of <code>sine</code> terms (traditionally the B terms). The
-                first element (index 0) should be set to zero (and will be
-                ignored) since this term does not exist in the Fourier series.
-                The second element (index 1) represents the fundamental
-                frequency. The third element represents the first overtone, and
-                so on.
+                of <code>sine</code> terms. The first element (index 0) does
+                not existe in the Fourier series. Implementations MUST set it
```
Comment (rtoy) [inline index.html]: s/existe/exist/

**Pull request merged**